File: Business/src/main/java/name/benjaminabbitt/evented/BusinessLogic/Bookmarks/DeletedBookmark.java
package name.benjaminabbitt.evented.BusinessLogic.Bookmarks;

import name.benjaminabbitt.evented.core.Evented;
import name.benjaminabbitt.evented.bookmarks.Bookmarks;

import java.util.stream.Stream;

public class DeletedBookmark implements Bookmark {
    public DeletedBookmark(ExtantBookmark old) {

    }

    @Override
    public Bookmark handle(Bookmarks.BookmarkDeleted bookmarkDeleted) throws InvalidState {
        throw new InvalidState();
    }

    @Override
    public Stream<Evented.EventPage.Builder> handle(Bookmarks.DeleteBookmark deleteBookmark) throws InvalidState {
        throw new InvalidState();
    }

    @Override
    public Bookmark handle(Bookmarks.BookmarkCreated bookmarkCreated) throws InvalidState {
        throw new InvalidState();
    }

    @Override
    public Stream<Evented.EventPage.Builder> handle(Bookmarks.CreateBookmark createBookmark) throws InvalidState {
        throw new InvalidState();
    }

}

File: BusinessLogic/src/main/kotlin/name/benjaminabbitt/wwwwwh/url/businesslogic/bookmarks/statemachine/states/DeletedBookmark.kt
package name.benjaminabbitt.wwwwwh.url.businesslogic.bookmarks.statemachine.states

import name.benjaminabbitt.evented.bookmarks.Bookmarks.*
import name.benjaminabbitt.evented.core.Evented
import name.benjaminabbitt.wwwwwh.url.businesslogic.bookmarks.statemachine.InvalidState
import org.slf4j.LoggerFactory
import java.util.*
import java.util.stream.Stream

class DeletedBookmark(id: UUID, old: ExtantBookmark?) : Bookmark(id, LoggerFactory.getLogger(DeletedBookmark::class.java)) {
    @Throws(InvalidState::class)
    override fun handle(bookmarkDeleted: BookmarkDeleted): Bookmark {
        throw InvalidState()
    }

    @Throws(InvalidState::class)
    override fun handle(deleteBookmark: DeleteBookmark): Stream<Evented.EventPage.Builder> {
        throw InvalidState()
    }

    @Throws(InvalidState::class)
    override fun handle(bookmarkCreated: BookmarkCreated): Bookmark {
        throw InvalidState()
    }

    @Throws(InvalidState::class)
    override fun handle(createBookmark: CreateBookmark): Stream<Evented.EventPage.Builder> {
        throw InvalidState()
    }
}